src/application-container/ApplicationConceptContext.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The ApplicationConceptContext is used to define an overarching concept that
+ * the application relates to at any given time. The components provided
+ * through this context will be rendered automatically by the framework
+ * to ensure that the concept is identifiable at any given time through the
+ * application's workflows.
+ */
+const ApplicationConceptContext = React.createContext({});
+
+/**
+ * Custom hook to retrieve the ApplicationConceptContext value. Presence of a
+ * value is not validated, as a provider is not required for all applications.
+ * @returns The retrieved ApplicationConceptContext value.
+ */
+const useApplicationConcept = () => {
+  const contextValue = React.useContext(ApplicationConceptContext);
+  return contextValue;
+};
```

Comment:
personal preference
```suggestion
const useApplicationConcept = () => React.useContext(ApplicationConceptContext);
```

Reply:
It's a good preference. Doesn't break the rules of hooks and looks a lot better.